pyccel/ast/builtin_methods/list_methods.py

```diff
@@ -83,7 +84,82 @@ def append_argument(self):
         """
         return self._append_arg
 
-#==============================================================================
+class ListInsert(PyccelInternalFunction):
+    """
+    Represents a call to the .insert() method.
+
+    Represents a call to the .insert() method of an object with a list type,
+    which inserts a given element at a given index in a list. This method returns `None`.
+    The insert method is called as follows:
+
+    >>> a = [2, 3, 4]
+    >>> a.insert(1, 1)
+    >>> print(a)
+    [1, 2, 3, 4]
```

Comment:
Did you run this test or just type it out? Python is zero indexed so shouldn't the result be:

```suggestion
    >>> a = [2, 3, 4]
    >>> a.insert(1, 1)
    >>> print(a)
    [2, 1, 3, 4]
```
?

Reply:
my bad, an oversight since i was thinking of inserting in the "first" position and got mixed up with index 1, will fix in the next commit

Reply:
fixed in 5351255e5d429627252ffe3fa7d18707a789db5f